cmd/loop/loopout.go

```diff
@@ -213,3 +239,41 @@ func loopOut(ctx *cli.Context) error {
 
 	return nil
 }
+
+// calculateMaxMinerFee validates the max miner fee set by cli is sane.
+func calculateMaxMinerFee(ctx *cli.Context,
+	amt btcutil.Amount) (btcutil.Amount, error) {
+
+	useSatoshi := ctx.IsSet("max_sweep_fee")
+	usePercent := ctx.IsSet("max_sweep_fee_percent")
+
+	// Check max sweep fee is set either in the format of satoshi or
+	// percent.
+	if useSatoshi && usePercent {
+		return 0, fmt.Errorf(
+			"either %s or %s should be set, not both",
+			"max_sweep_fee", "max_sweep_fee_percent",
+		)
+	}
+
+	if useSatoshi {
+		fee := btcutil.Amount(ctx.Uint64("max_sweep_fee"))
+		if fee >= amt {
+			return 0, fmt.Errorf("max_sweep_fee (%d) cannot be "+
+				"greater than total swap amount (%d)", fee, amt,
+			)
+		}
+		return btcutil.Amount(ctx.Uint64("max_sweep_fee")), nil
+	}
+
+	// Validate percent is sane.
+	percent := ctx.Float64("max_sweep_fee_percent")
+	if percent < 0 || percent >= 1 {
+		return 0, fmt.Errorf(
+			"max_sweep_fee_percent is invalid, must be in "+
+				"range(0, 1), instead got %f", percent,
+		)
+	}
+
+	return amt.MulF64(percent), nil
```

Comment:
It wasn't clear to me that this is returning a zero value in the case where our percentage flag isn't set. 

I would add a `if usePercent` here and then validate that it `!=0` and then default to returning the prepay if neither option were set. I think it's ok to pass the prepay value in, even if it's just passed back out if no custom values are set.

Reply:
Switch to case for clarity. Passing prepay or a custom value doesn't worth the gain as it would only be used at the end of the function, then might as well use it outside the func. Plus the purpose of the function is to calculate the fee, returning zero seems a natural choice if it's not set.